vuu-ui/packages/vuu-shell/test/layout-management/useLayoutManager.test.tsx

```diff
@@ -0,0 +1,209 @@
+  screenshot: "test-screenshot",
+  user: "test-user",
+  created: "test-date",
+  id: "test-id",
+};
+
+const initialApplicationJSON = vi.hoisted(() => ({
+  layout: defaultLayout,
+}));
+
+vi.stubEnv("LOCAL", "true");
+
+vi.mock(
+  "../../src/persistence-management/LocalPersistenceManager",
+  async () => {
+    const MockPersistenceManager = vi.fn();
+    MockPersistenceManager.prototype.loadMetadata = vi.fn();
+    MockPersistenceManager.prototype.loadApplicationJSON = vi.fn();
+    MockPersistenceManager.prototype.saveApplicationJSON = vi.fn();
+    MockPersistenceManager.prototype.createLayout = vi.fn();
+    MockPersistenceManager.prototype.updateLayout = vi.fn();
+    MockPersistenceManager.prototype.deleteLayout = vi.fn();
+    MockPersistenceManager.prototype.loadLayout = vi.fn();
+
+    return { LocalPersistenceManager: MockPersistenceManager };
+  }
+);
+
+vi.mock("../../src/persistence-management/defaultApplicationJson", async () => {
+  const actual = await vi.importActual<
+    typeof import("../../src/persistence-management/defaultApplicationJson")
+  >("../../src/persistence-management/defaultApplicationJson");
+  return {
+    ...actual,
+    loadingApplicationJson: initialApplicationJSON,
+  };
+});
+
+vi.mock("@finos/vuu-popups", async () => {
+  const actual = await vi.importActual<typeof import("@finos/vuu-popups")>(
+    "@finos/vuu-popups"
+  );
+  const mockNotify = vi.fn();
+  return {
+    ...actual,
+    useNotifications: vi.fn(() => ({ notify: mockNotify })),
+  };
+});
+
+vi.mock("@finos/vuu-layout", async () => {
+  const actual = await vi.importActual<typeof import("@finos/vuu-layout")>(
+    "@finos/vuu-layout"
+  );
+  return {
+    ...actual,
+    isLayoutJSON: vi.fn(),
+    resolveJSONPath: vi.fn(),
+  };
+});
+
+const wrapper = ({ children }) => (
+  <LayoutManagementProvider>{children}</LayoutManagementProvider>
+);
+
+describe("LayoutManagementProvider", () => {
+  let persistence: PersistenceManager;
+
+  beforeEach(() => {
+    persistence = new LocalPersistenceManager();
+    vi.mocked(persistence.loadMetadata).mockResolvedValueOnce([]);
+    vi.mocked(persistence.loadApplicationJSON).mockResolvedValueOnce(
+      initialApplicationJSON
+    );
+    vi.spyOn(global.console, "error");
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  it("calls loadMetadata and loadApplicationJSON on mount", () => {
+    const { result } = renderHook(() => useLayoutManager(), { wrapper });
+
+    expect(persistence.loadMetadata).toHaveBeenCalled();
+    expect(persistence.loadApplicationJSON).toHaveBeenCalled();
+    expect(result.current.applicationJson).toBe(initialApplicationJSON);
+  });
+
+  describe("saveApplicationLayout", () => {
+    it("calls saveApplicationJSON when layout is valid", () => {
+      const { result } = renderHook(useLayoutManager, { wrapper });
+
+      vi.mocked(persistence.saveApplicationJSON).mockResolvedValueOnce();
+      vi.mocked(isLayoutJSON).mockReturnValue(true);
```

Comment:
Do we have to mock the result of `isLayoutJSON`? I think it would be best to test that the function is working as expected, so would that mean writing tests for the utils file?

Reply:
I think it makes sense to mock it here as the test should only be concerned with testing the functionality of the hook. I would test that separately together with the other functionalities from utils file

Reply:
I have added a couple of test cases for it in a separate test file